tests/integration/resource_update_test.go

```diff
@@ -0,0 +1,265 @@
+//go:build integration
+// +build integration
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	sambaoperatorv1alpha1 "github.com/samba-in-kubernetes/samba-operator/api/v1alpha1"
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/kube"
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/poll"
+	"github.com/samba-in-kubernetes/samba-operator/tests/utils/smbclient"
+)
+
+var (
+	// waitForSyncTime is 1.5 times the default Kubelet syncFrequency
+	waitForSyncTime = 90 * time.Second
+)
+
+type SmbShareUpdateSuite struct {
+	suite.Suite
+
+	commonSources   []kube.FileSource
+	smbShareSources []kube.FileSource
+	shareName       string
+	testAuths       []smbclient.Auth
+	destNamespace   string
+
+	// cached values
```

Comment:
Style: I find those comment redundant. This type is self-explanatory; no need for those extra comments. IMHO a single comment before the struct declaration is more then enough here.

Reply:
You're right. I had been doing that as a reminder to myself which variables are meant as "inputs" vs stuff that shouldn't be treated as an input. Because all the types are private I wanted that reminder to myself. However if I want to be serious about that it may be better to use another technique like a newFoo method or even an embedded struct, where the embedded struct is the "input" values.  Regardless of what we do in the future I agree with you that the comment is not hepful and I'll try to remove them in the future.